### DISCUSSION TITLE
What are the risks of using legacy decorators?

### DISCUSSION THREAD

**Discussion by alexfoxy** [mobxjs/mobx#3929] [Q&A] [answered]
We have a very large code base that we want to upgrade to mobx 6. We've been using the legacy style of decorators for 5 years which has been working fine but now we're looking to migrate. However we've run into some issues with chained decorators not compiling correctly and I was looking at the option of just continuing to use the legacy style.

I note in the docs it says:
> We do not recommend codebases to use TypeScript / Babel legacy decorators since they well never become an official part of the language, but you can still use them. It does require a specific setup for transpilation:

Will the legacy style of decorators ever be deprecated, and are there any risks of not adopting the new style, for example library compatibility?

Thanks
Alex

**Comment by urugator** [comment, marked as answer]
It's very likely that any new major will only support https://github.com/tc39/proposal-decorators with all it's pros and cons.

See https://github.com/mobxjs/mobx/issues/3796
